senders/pushover/pushover.go

```diff
@@ -71,27 +74,94 @@ func (sender *Sender) makePushoverMessage(events moira.NotificationEvents, conta
 	return pushoverMessage
 }
 
-func (sender *Sender) buildMessage(events moira.NotificationEvents, throttled bool) string {
-	var message bytes.Buffer
-	for i, event := range events {
-		if i > printEventsCount-1 {
-			break
-		}
-		message.WriteString(fmt.Sprintf("%s: %s = %s (%s to %s)", event.FormatTimestamp(sender.location), event.Metric, event.GetMetricValue(), event.OldState, event.State))
+func (sender *Sender) buildMessage(events moira.NotificationEvents, throttled bool, trigger moira.TriggerData) string {
+	var message strings.Builder
+
+	desc := trigger.Desc
+	htmlDesc := string(blackfriday.Run([]byte(desc)))
+	htmlDescLen := len([]rune(htmlDesc))
+	charsForHTMLTags := htmlDescLen - len([]rune(desc))
+
+	eventsString := sender.buildEventsString(events, -1, throttled)
+	eventsStringLen := len([]rune(eventsString))
+
+	if htmlDescLen+eventsStringLen <= msgLimit {
+		// Use both completely
+		message.WriteString(htmlDesc)
+		message.WriteString(eventsString)
+		return message.String()
+	}
+
+	if htmlDescLen > msgLimit/2 && eventsStringLen > msgLimit/2 {
+		// Trim both desc and events to half the message size each
+		desc = desc[:msgLimit/2-charsForHTMLTags-10] + "...\n"
+		htmlDesc = string(blackfriday.Run([]byte(desc)))
+		eventsString = sender.buildEventsString(events, msgLimit/2, throttled)
+
+	} else if htmlDescLen > msgLimit/2 {
+		// Trim the desc to the chars left after using the whole events string
+		charsForDesc := msgLimit - eventsStringLen
+		desc = desc[:charsForDesc-charsForHTMLTags-10] + "...\n"
+		htmlDesc = string(blackfriday.Run([]byte(desc)))
+
+	} else if eventsStringLen > msgLimit/2 {
+		// Trim the events string to the chars left after using the whole desc
+		charsForEvents := msgLimit - htmlDescLen
+		eventsString = sender.buildEventsString(events, charsForEvents, throttled)
+
+	} else {
+		// Trim both desc and events to half the message size each
+		desc = desc[:msgLimit/2-charsForHTMLTags-10] + "...\n"
+		htmlDesc = string(blackfriday.Run([]byte(desc)))
+		eventsString = sender.buildEventsString(events, msgLimit/2, throttled)
+
+	}
```

Comment:
I merged the lines 88...93 as a wrapping if block, so that it doesn't become an empty block. So from your example, if (1) is not true then consider the cases (2-4).